File: Java/Python/Ejercicio 2.py
# Python3 program to check whether two 
# strings are Permutations of each other

# function to check whether two strings 
# are Permutation of each other */
def arePermutation(str1, str2):
	n1 = len(str1)
	n2 = len(str2)
	
	if (n1 != n2):
		return False
	
	a = sorted(str1)
	str1 = " ".join(a)
	b = sorted(str2)
	str2 = " ".join(b)

	for i in range(0, n1, 1):
		if (str1[i] != str2[i]):
			return False

	return True

if __name__ == '__main__':
	print("INGRESA DOS CADENAS ")
	str1 = input()
	str2 = input() 

	if (arePermutation(str1, str2)):
		print("Yes")
	else:
		print("No")

File: Java/Ejercicio2.java
/* Java program to check whether two strings are Permutations of each other */

import java.util.*;
class GfG{ 

	static boolean arePermutation(String str1, String str2) { 
		int n1 = str1.length(); 
		int n2 = str2.length(); 

		if (n1 != n2) 
			return false; 

		char ch1[] = str1.toCharArray();
		char ch2[] = str2.toCharArray();

		Arrays.sort(ch1); 
		Arrays.sort(ch2); 

		for (int i = 0; i < n1; i++) {
			if (ch1[i] != ch2[i])
				return false; 
		}
		return true; 
	} 

	public static void main(String[] args)  { 
        Scanner lectura = new Scanner (System.in);
        System.out.print("Ingresa una palabra: ");
        String str1 = lectura.next();
        String str2 = lectura.next();
		if (arePermutation(str1, str2)) 
		    System.out.println("Yes"); 
		else
		    System.out.println("No"); 
	}
} 
